sections/01_structured_workflow.qmd

```diff
@@ -1,21 +1,148 @@
+journey to good coding practice, don't feel overwhelmed by all the rules and 
+conventions. You don't need a perfect workflow and codebase from the beginning.
+Start with one guideline and build from there. Writing reproducible and legible code will become a habit over time and you will notice that this does not make coding more burdonsome but actually faster and more fun and rewarding.
+
+### Have a consistent coding style
+
+Consistency improves readability and maintainability. It helps others to quickly understand your project's logic and workflows. 
+
+#### Follow a style guide
+
+Every programming language has a style guide for things like
+indentation, spacing and naming conventions for variables and functions. Check the style guide for the languages you use to get an overview [^1].
+
+The cool thing is, that there are tools to help you enforce a consistent style across a codebase:
+
+- **Formatters**: Can auto-format your script to eliminate inconsistencies.
+- **Linters**: Can analyze your code and list areas for improvement.
+
+Using such tools is an easy win to improve your code base and it is very satisfying to
+see your whole script automatically transforming into a good practice style.
+
+::: {.callout-tip}
+
+## Using formatters and Linters
+
+The details of this depend on your combination of IDE and programming language. Just search for the right formatters and linters for your setup. 
+
+##### R and RStudio
+
+- **Auto-format**: Open the command palette (`Ctrl/Cmd + Shift + P`) and search for "format". Use "Reformat Current Document" to auto-format your code. Toggle "Reformat documents on save" for convenience. You can also chose your own "Code formatter" but the default option "Styler", which applies the tidyverse style guide is already a very good option.
+- **Lint**: Install the `lintr` package and search for "lint" options in the command palette. Use "Lint current file" or "Lint current package" to list style problems.
+
+#### Python and VScode
+
+<!-- todo: this still needs to be written -> but maybe it is anyways already too long and too specific? -->
+- **Auto-format**: Open the command palette (`Ctrl/Cmd + Shift + P`) and search for "Format Document". You can also enable "Format on Save" in VS Code settings and select a default formatter in the VS Code settings.
+- **Lint**: Install a linter like `pylint` or `flake8` and configure it in VS Code settings.
+
+:::
+  
+#### Have a naming convention
+
+Use clear and descriptive names for files, variables, functions, and modules. 
+The goal of a good naming conventions is that it is immediately clear to the reader
+what is behind any file or object.
+To learn about good practice names for your files, please refer to the section on project structure. For objects in your codebase you can follow these tips:
+
+- **Concise and Descriptive**: Variable names are usually nouns, and function names are verbs.
+- **Avoid Conflicts**: Don't use names of existing variables or functions.
```

Comment:
This might be language specific but I would argue in Julia if you are developing a new method for a well known/defined function *e.g.,* calculating mean for an object of type e.g., SDMlayer then it is better to redefine a function with a new method (mean(T::SDMlayer)) as opposed to creating a mean_SDMlayer() function. 

I'm throwing this in as a comment but given the degree of language specificity I think it is worth leaving out

Reply:
I think that works well. Short and sweet. We could even leave out the e.g. but lets keep that for now)